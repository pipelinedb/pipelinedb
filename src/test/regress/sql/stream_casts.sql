SET debug_sync_stream_insert = 'on';

CREATE CONTINUOUS VIEW test_stream_casts0 AS SELECT k::text, SUM(v::float8) AS fsum, SUM(i::int8) AS isum, COUNT(*) FROM stream_casts_stream GROUP BY k;

CREATE CONTINUOUS VIEW test_stream_casts1 AS SELECT k::integer, SUM(v::float4) AS fsum, SUM(i::int4) AS isum, COUNT(*) FROM stream_casts_stream GROUP BY k;

ACTIVATE test_stream_casts0, test_stream_casts1;

INSERT INTO stream_casts_stream (k, v, i) VALUES ('001', 1.224, 1002);
INSERT INTO stream_casts_stream (k, v, i) VALUES ('102', 1e15, -100);
INSERT INTO stream_casts_stream (k, v, i) VALUES ('144', -1e-3, 1);

DEACTIVATE test_stream_casts0, test_stream_casts1;

SELECT * FROM test_stream_casts0 ORDER BY k;
SELECT * FROM test_stream_casts1 ORDER BY k;

DROP CONTINUOUS VIEW test_stream_casts0;
DROP CONTINUOUS VIEW test_stream_casts1;

