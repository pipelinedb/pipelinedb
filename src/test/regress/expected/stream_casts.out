SET debug_sync_stream_insert = 'on';
CREATE CONTINUOUS VIEW test_stream_casts0 AS SELECT k::text, SUM(v::float8) AS fsum, SUM(i::int8) AS isum, COUNT(*) FROM stream_casts_stream GROUP BY k;
CREATE CONTINUOUS VIEW test_stream_casts1 AS SELECT k::integer, SUM(v::float4) AS fsum, SUM(i::int4) AS isum, COUNT(*) FROM stream_casts_stream GROUP BY k;
ACTIVATE test_stream_casts0, test_stream_casts1;
INSERT INTO stream_casts_stream (k, v, i) VALUES ('001', 1.224, 1002);
INSERT INTO stream_casts_stream (k, v, i) VALUES ('102', 1e15, -100);
INSERT INTO stream_casts_stream (k, v, i) VALUES ('144', -1e-3, 1);
INSERT INTO stream_casts_stream (k, v, i) VALUES ('042', 42.42, 0), ('42', '42.42', 1), ('144', 1, 2);
DEACTIVATE test_stream_casts0, test_stream_casts1;
SELECT * FROM test_stream_casts0 ORDER BY k;
  k  | fsum  | isum | count 
-----+-------+------+-------
 001 | 1.224 | 1002 |     1
 042 | 42.42 |    0 |     1
 102 | 1e+15 | -100 |     1
 144 | 0.999 |    3 |     2
 42  | 42.42 |    1 |     1
(5 rows)

SELECT * FROM test_stream_casts1 ORDER BY k;
  k  | fsum  | isum | count 
-----+-------+------+-------
   1 | 1.224 | 1002 |     1
  42 | 84.84 |    1 |     2
 102 | 1e+15 | -100 |     1
 144 | 0.999 |    3 |     2
(4 rows)

DROP CONTINUOUS VIEW test_stream_casts0;
DROP CONTINUOUS VIEW test_stream_casts1;
